akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -412,6 +412,14 @@ object Source {
   def lazily[T, M](create: () ⇒ Source[T, M]): Source[T, Future[M]] =
     Source.fromGraph(new LazySource[T, M](create))
 
+  /**
+   * Creates a `Source` from supplied future factory that is not called until downstream demand. When source gets
+   * materialized the materialized future is completed with the value from the factory. If downstream cancels or fails
+   * without any demand the create factory is never called and the materialized `Future` is failed.
+   */
+  def lazilyAsync[T](create: () ⇒ CompletionStage[T]): Source[T, Future[NotUsed]] =
```

Comment:
This should work on Scala.concurrent.Future :)

Reply:
Makes sense, also done.